include/glow/Support/Log.h

```diff
@@ -0,0 +1,301 @@
+    logStr.append(") (");                                                      \
+    logStr.append(src5->getKindName());                                        \
+    logStr.append(", ");                                                       \
+    logStr.append(src5->getName());                                            \
+    logStr.append(") }\n");                                                    \
+    getLogContext().addLogContent(logStr);                                     \
+  }
+
+/// Logs the nodes replacement.
+#define LOG_NODE_REPLACE(oldNode, newNode)                                     \
+  if (dumpCompilationLogOpt) {                                                 \
+    std::string logStr = "";                                                   \
+    logStr.append("[FULL SCOPE: ");                                            \
+    logStr.append(getNode()->getParent()->getLogContext().getFullScopeName()); \
+    logStr.append(" ]");                                                       \
+    logStr.append(" --- REPLACE { ( ");                                        \
+    logStr.append(oldNode->getKindName());                                     \
+    logStr.append(", ");                                                       \
+    logStr.append(oldNode->getName());                                         \
+    logStr.append(" ) <== ( ");                                                \
+    logStr.append(newNode->getKindName());                                     \
+    logStr.append(", ");                                                       \
+    logStr.append(newNode->getName());                                         \
+    logStr.append(" ) }\n");                                                   \
+    getNode()->getParent()->getLogContext().addLogContent(logStr);             \
+  }
+
+/// Logs the nodes deletion.
+#define LOG_NODE_DELETE(deletedNode)                                           \
+  if (dumpCompilationLogOpt) {                                                 \
+    std::string logStr = "";                                                   \
+    logStr.append("[FULL SCOPE: ");                                            \
+    logStr.append(getLogContext().getFullScopeName());                         \
+    logStr.append(" ]");                                                       \
+    logStr.append(" --- DELETE { (");                                          \
+    logStr.append(deletedNode->getKindName());                                 \
+    logStr.append(", ");                                                       \
+    logStr.append(deletedNode->getName());                                     \
+    logStr.append(") }\n");                                                    \
+    getLogContext().addLogContent(logStr);                                     \
+  }
+
+/// Logs a new log scope.
+#define LOG_SCOPE(ctx, name) ScopedLogBlock __scope__(ctx, name);
```

Comment:
Since the name of the scope is fixed to `__scope__`, one can have only one `ScopedLogBlock ` object in any concrete scope, otherwise there will be name conflicts. This is probably intentional. So, I'm just checking.

Reply:
Can have another macro with named scope